Framework/Interaction/Rest/Customer.php

```diff
@@ -170,4 +186,86 @@ public function deleteCertificate(
         $this->validateResult($result, $request);
         return $this->formatResult($result);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateCustomer(
+        $customer,
+        $isProduction = null,
+        $scopeId = null,
+        $scopeType = \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+    )
+    {
+        // Client must be retrieved before any class from the /avalara/avataxclient/src/Models.php file is instantiated.
+        $client = $this->getClient($isProduction, $scopeId, $scopeType);
+        $client->withCatchExceptions(false);
+        $customerModel = $this->buildCustomerModel($customer, $scopeId, $scopeType); // Instantiates an Avalara class.
+
+        $response = null;
+
+        try {
+            $response = $client->updateCustomer(
+                $this->config->getCompanyId($scopeId, $scopeType),
+                $this->customerHelper->getCustomerCode($customer->getId(), null, $scopeId),
+                $customerModel
+            );
+        } catch (\GuzzleHttp\Exception\ClientException $clientException) {
+            // Validate the response; pass the customer id for context in case of an error.
+            $this->handleException($clientException, $this->dataObjectFactory->create(['customer_id' => $customer->getId()]));
+        }
+
+        return $this->formatResult($response);
+    }
+
+    /**
+     * Given a Magento customer, build an Avalara CustomerModel for request.
+     *
+     * @param \Magento\Customer\Api\Data\CustomerInterface $customer
+     * @param null $scopeId
+     * @param string $scopeType
+     * @return \Avalara\CustomerModel
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    protected function buildCustomerModel(
+        \Magento\Customer\Api\Data\CustomerInterface $customer,
+        $scopeId = null,
+        $scopeType = \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+    )
+    {
+        /** @var \Avalara\CustomerModel $customerModel */
+        $customerModel = $this->customerModelFactory->create();
+
+        $customerModel->customerCode = $this->customerHelper->getCustomerCode($customer->getId(), null, $scopeId);
+        $customerModel->name = $customer->getFirstname() . ' ' . $customer->getLastname();
```

Comment:
It's faster in PHP and easier to read if you use string interpolation instead of concatenation. 

```suggestion
        $customerModel->name = "{$customer->getFirstname()} {$customer->getLastname()}";
```

Reply:
I would argue this is much harder to read.... but ok. 😄